core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -223,9 +223,13 @@ public void testToQuery() throws IOException {
 
         Query expectedQuery = createExpectedQuery(testQuery, context);
         Query actualQuery = testQuery.toQuery(context);
-        assertThat(actualQuery, instanceOf(expectedQuery.getClass()));
-        assertThat(actualQuery, equalTo(expectedQuery));
-        assertLuceneQuery(testQuery, actualQuery, context);
+        if (expectedQuery == null) {
```

Comment:
can we add a small comment just to remind why we might have null here? I forget all the time :)